query/src/storages/memory/memory_table.rs

```diff
@@ -71,17 +73,44 @@ impl Table for MemoryTable {
         &self.table_info
     }
 
+    fn benefit_column_prune(&self) -> bool {
+        true
+    }
+
     async fn read_partitions(
         &self,
         ctx: Arc<QueryContext>,
-        _push_downs: Option<Extras>,
+        push_downs: Option<Extras>,
     ) -> Result<(Statistics, Partitions)> {
         let blocks = self.blocks.read();
 
-        let rows = blocks.iter().map(|block| block.num_rows()).sum();
-        let bytes = blocks.iter().map(|block| block.memory_size()).sum();
+        let statistics = if let Some(Extras {
```

Comment:
maybe use match is better?

Reply:
Yes, match is better for future work. let me fix it.